diff --git glob/glob.c glob/glob.c
index f3911bc..241c1c2 100644
--- glob/glob.c
+++ glob/glob.c
@@ -208,7 +208,7 @@ my_realloc (p, n)
 #endif /* __GNU_LIBRARY__ || __DJGPP__ */
 
 
-#if !defined __alloca && !defined __GNU_LIBRARY__
+#if !defined __alloca && defined __GNU_LIBRARY__
 
 # ifdef	__GNUC__
 #  undef alloca
@@ -231,7 +231,7 @@ extern char *alloca ();
 
 #endif
 
-#ifndef __GNU_LIBRARY__
+#ifdef __GNU_LIBRARY__
 # define __stat stat
 # ifdef STAT_MACROS_BROKEN
 #  undef S_ISDIR
